tests/kernel/sched/preempt/src/main.c

```diff
@@ -96,17 +106,21 @@ void wakeup_src_thread(int id)
 			     "worker thread %d not pending?", i);
 	}
 
+	/* Wake the src worker up */
 	last_thread = NULL;
 	k_sem_give(&worker_sems[id]);
 
+	while (do_sleep
+	       && !(worker_threads[id].base.thread_state & _THREAD_PENDING)) {
```

Comment:
Won't this test immediately break when you submit the k_sleep re-write for v1.13?

Reply:
Likely, yes.  :)  But I'm whiteboxing the thread states here anyway and this is the easiest way to make sure the thread is blocked on the semaphore again before going to the next test case.

Reply:
Sounds good.  Removing my -1.